### PR TITLE
[execution] change HashMap to BTreeMap in VMChangeSet

### PR DESCRIPTION
**Pull request by ibalajiarun**

### CODE REVIEW

**Review by igor-aptos — APPROVED** [review]

File: docker/builder/tools.Dockerfile [hunk 1]
```diff
@@ -11,7 +11,7 @@ RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
     wget \
     curl \
     perl-base=5.32.1-4+deb11u1 \
-    libtinfo6=6.2+20201114-2+deb11u1 \
+    libtinfo6=6.2+20201114-2+deb11u2 \
```
Comment (sitalkedia) [inline docker/builder/tools.Dockerfile:14]: unrelated change?

Comment (ibalajiarun) [reply docker/builder/tools.Dockerfile:14]: without it, image building is broken.

**Review by sitalkedia — APPROVED** [review]
